### PR TITLE
Various cleanups to backends code.

### PR DESCRIPTION
**Pull request by anntzer**
Removed some debug statements and old comments; whitespace / line
wrapping (to 79 characters).

(FWIW the debug code in backend_gtkcairo is apparently broken -- it occasionally refers to a function `fn_name` which doesn't exist (it's `_fn_name`))

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/whats_new.rst if major new feature
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_tkagg.py [hunk 1]
```diff
@@ -731,11 +730,10 @@ def set_message(self, s):
...
     def draw_rubberband(self, event, x0, y0, x1, y1):
         height = self.canvas.figure.bbox.height
-        y0 =  height-y0
-        y1 =  height-y1
-        try: self.lastrect
-        except AttributeError: pass
-        else: self.canvas._tkcanvas.delete(self.lastrect)
+        y0 = height - y0
+        y1 = height - y1
+        if hasattr(self, "lastrect"):
```
Comment (tacaswell) [inline lib/matplotlib/backends/backend_tkagg.py]: This is a hot-path for UI,  Under normal usage I think we expect this to exist for often that in does not so the `try..except` pattern is marginally faster.

```python

def tester(obj):
    if hasattr(obj, 'string'):
        pass

def tryer(obj):
    try:
        obj.string
    except:
        pass


class A:
    def __init__(self):
        self.string = 'foo'

class B:
    ...
```

```python
In [61]: %timeit tester(1)
459 ns ± 3.11 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [62]: %timeit tryer(1)
524 ns ± 0.955 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [63]: 
In [64]: a = A()

In [65]: %timeit tryer(a)
123 ns ± 0.202 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [66]: %timeit tester(a)
194 ns ± 0.281 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [67]: 
In [68]: b = B()

In [69]: %timeit tryer(b)
488 ns ± 1.11 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [70]: %timeit tester(b)
423 ns ± 2.14 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [74]: 
```

**Comment by tacaswell** [comment]
:+1: modulo the hot-path concern.

File: lib/matplotlib/backends/backend_tkagg.py (hunk 1 above)
Comment (anntzer) [inline lib/matplotlib/backends/backend_tkagg.py]: The following patch
```
diff --git a/lib/matplotlib/backends/backend_tkagg.py b/lib/matplotlib/backends/backend_tkagg.py
index af0ed62f9..4d212efe8 100644
--- a/lib/matplotlib/backends/backend_tkagg.py
+++ b/lib/matplotlib/backends/backend_tkagg.py
@@ -736,7 +736,14 @@ class NavigationToolbar2TkAgg(NavigationToolbar2, Tk.Frame):
         try: self.lastrect
         except AttributeError: pass
         else: self.canvas._tkcanvas.delete(self.lastrect)
+        from time import perf_counter
+        start = perf_counter()
+        elapsed_dummy = perf_counter() - start
+        start = perf_counter()
         self.lastrect = self.canvas._tkcanvas.create_rectangle(x0, y0, x1, y1)
+        elapsed_draw = perf_counter() - start
+        elapsed_corrected = elapsed_draw - elapsed_dummy
+        print(f"create_rectangle took {elapsed_corrected * 1e6}us")
```
shows that the call to create_rectangle that's just below takes ~100-300us, so I think worrying about 50ns for the sake of clarity is totally overkill.

**Comment by tacaswell** [comment]
The check is not the slowest thing in this code path, so :+1:

**Pull request merged**